src/3d/mesh/qgsmesh3dgeometry_p.cpp

```diff
@@ -144,7 +144,7 @@ static QByteArray createDatasetIndexData( const QgsTriangularMesh &mesh, const Q
   for ( int i = 0; i < trianglesCount; ++i )
   {
     int nativeFaceIndex = mesh.trianglesToNativeFaces()[i];
-    const bool isActive = mActiveFaceFlagValues.active( nativeFaceIndex );
+    const bool isActive = mActiveFaceFlagValues.active().isEmpty() || mActiveFaceFlagValues.active( nativeFaceIndex );
```

Comment:
it does not harm, but not sure if 100% required https://github.com/qgis/QGIS/blob/master/src/core/mesh/qgsmeshdataprovider.cpp#L292